### DISCUSSION TITLE
I don't want algorithmic feed

### DISCUSSION THREAD

**Discussion by ericoporto** [community/community#13131] [Other Feature Feedback, Questions, & Ideas] [answered]
I just want a feed on things that matters:

- releases
- issues
- PRs

Also the ability to mute things I don't want to be notified anymore.

Also the new feed crashes on Chrome mobile, occasionally.

We already lost the recent activities dashboard, don't kill the feed too.

**Comment by Owen-Choi** [comment]
👍

**Reply by jillesvangurp** [reply]
Just adding my 2 cents. Facebook is dead to me. Twitter is dead to me. The Linkedin feed (also owned by MS) is dead to me. I use it to mainly ignore recruiter spam, check up on the occasional profile, and ignore the feed.

Most of my Facebook friends have just left the platform. I check in once every few weeks. There's nothing there anymore. Twitter is just an endless cycle of mostly random stuff. Twitter used to be about real-time, happening right now, seconds ago. Not anymore. They threw the baby away with bathwater with that one. It's useless for that now. 

The reason: algorithmic feeds don't actually work. The feeds just get hijacked by the most boring and dull people ever: influencers. We don't need Github influencers.

When I check the Github feed, I expect to see updates from people and projects I follow. Any other feed is useless to me. So, please don't waste it with random stuff we never asked for. It's actually nice as it is. Don't mess it up please.

**Reply by NearJonathanCraig** [reply]
I could not agree more with this sentiment.  Feeds are so 2000-and-late, and so is this sentence.

**Reply by ankush981** [reply]
Yes! The Feed shouldn't be forced-fed (pun intended) to me. Rather, I want my feed to feed me stuff only from the projects I follow or have starred. I don't want "intelligent" suggestions, thank you very much. 🙄

**Reply by twojstaryzdomu** [reply]
The very nature of a site like github, where user-generated content is based on merit, rules out the idea of influencers. It may work for another site where users can be fooled easily by who talks loudest but not here.

**Reply by MichaelWheldon** [reply]
@twojstaryzdomu theres a whole world of code influencers out there already. So far I’ve managed to create ways to filter them out of my life. This feed would be an avenue to amplify these people / businesses, thus the concerns.

**Reply by twojstaryzdomu** [reply]
I must have been happily living under a rock away from those influencers then. 
That's likely the long term plan, to advertise services. If any conclusions are to be drawn from how feeds were implemented elsewhere (not excluding other services hoovered up by Microsoft), it is gradually littering them with ads. Would that be the reason why no off switch has been made available?

**Reply by fezzopro** [reply]
I think GitHub is heading in the wrong direction with the feeds, soon it'll become like any other social media and start to annoy users with unwanted feeds.

**Reply by loneicewolf** [reply]
> Yes! The Feed shouldn't be forced-fed (pun intended) to me. Rather, I want my feed to feed me stuff only from the projects I follow or have starred. I don't want "intelligent" suggestions, thank you very much. roll_eyes

I know this was awhile ago; (2022) But that pun - ~~was~~ (`still`) is just flawless!

yea. That is all I had to say. 👍🏽 Have a great day everyone ^_^

**Comment by abathur** [comment]
Agree. I don't mind an algo-feed existing for people who do prefer it (or, for that matter, if our dash is ~widgetized), but I want predictable chronological feeds.

**Comment by Livinglist** [comment]
Me either, can we have an option to opt out "For You" tab?

**Reply by kotx** [reply]
They should put it under the "feature previews" so we can opt out

**Comment by dmadisetti** [comment]
+1 to muting people or events (or events by people)

-1 to the algorithmic feed. Just looking at it, most was people I don't know and don't care about interacting with projects that are not interesting. There's a lot of work to be done here if it is ever to be useful

Less noise not more please :)

**Reply by dmadisetti** [reply]
Did they just ship this feature 40 minutes ago?  Wow, we're fast 🚀

**Reply by ABSanthosh** [reply]
probably they already had this idea but still fast

**Reply by leon-thomm** [reply]
> most was people I don't know

well, that's kind of the point, isn't it

**Comment by lukesdm** [comment]
Yup. I thought we were over this. 

*Don't forget to like and subscribe*

**Comment by ABSanthosh** [comment]
I'd actually love this feature. For people who don't, maybe a toggle in settings to turn off the feed.

As Github have a lot of data on how we interact with certain projects, It might actually be helpful for us to discover similar projects

**Reply by Owen-Choi** [reply]
yeah. I think your opinion will be pretty good  solution

**Comment by Olyno** [comment]
The release / issues / pr feed is good for the "following" tab. On my side, I would like to discover more projects from people I don't know, or issues I could contribute. Currently, I just have a feed of people who are following me, which is not interesting for me. It's just the "following" tab but filled with my followers, and this is not interesting.

**Comment by bdresser** [comment]
Thanks for the feedback! We do have plans to experiment with the ability for the viewer to toggle between chronological and algorithmic. 

Regarding muting – do you find yourself looking for the ability to mute...
* An individual user?
* Activity on a project/repository? 
* A particular content type? 
* All of the above?

**Reply by ABSanthosh** [reply]
Github Actions! Please. That thing is filling up my feed so much that I just trigger it manually once in a while.

**Reply by jamieshark** [reply]
We have also considered the option of having various user-generated feed tabs that would only include content that you've selectively filtered and can also sort how you'd like. For example, a feed that only shows releases from certain repositories and sorted chronologically (or any other number of combinations you could create yourself). Are there certain filters or sorts that you'd prioritize being available?

**Reply by dmadisetti** [reply]
All of the above!

> An individual user?

I follow some people out of professional courtesy, I would love to be able to mute them (sorry not sorry). It would probably make me more likely to follow other people back. But I'd like to do it on a per event level. Some people like a lot of things, some people fork a lot of things- but I might be interested in the other events.

> Activity on a project/repository?

In particular can we mute repo likes. I think the granular "watching" criteria work otherwise.

> A particular content type?

I'm sure that would be nice. I would not use it if we had per user abilities though.

**Reply by ericoporto** [reply]
User or project/repository.

Honestly, I prefer to be in control, the feeling is the same of when Google killed the Reader, which made temporary have my own local rss feed, which then when that specific computer died I just abandoned news/blogs altogether. This change will probably make me abandon looking at the feed altogether. I will just manually visit the few projects I care about and close the tab and move on to other websites.

I really disliked when the recent activities dashboard got killed for no reason and now it's other feature getting killed. I am reasonably active here and try to track a bunch of things. Honestly, this is the biggest disappointment I ever felt with GitHub.

Things that are actually useful that have no investment. A board feature that doesn't suck, and can actually run kanban/scrum methods; epic issues so we can break features in smaller pieces of tasks; a way to support branches and tags on wikis (even just allow a gh repo view of them); a way to figure out which repositories you own that people are actually directing traffic and cloning without checking each one of them.

**Reply by panoply** [reply]
Posted before seeing this thread. Muting is an essential, dropping here for reference:
https://github.com/github/feedback/discussions/13282

**Reply by pauldps** [reply]
All of the above please. Also +1 for chronological feed.

**Reply by cuducos** [reply]
> We do have plans to experiment with the ability for the viewer to toggle between chronological and algorithmic.

I think that is actually terrible. It looks like a nice middle ground, but it actually is not:

* the [toxicity of gamification](https://github.com/github/feedback/discussions/13131#discussioncomment-2409431) put by @lukesdm would still be part of GitHub
* [privacy issues](https://github.com/github/feedback/discussions/13131#discussioncomment-2410835) (more data than needed being processed) as put by @bearlike would still be at place
* depending on the implementation, GitHub can force algorithmic feed (e.g. Twitter saves the preference for chronolopgical vs algorithmic feed in the client-side, so users with privacy and digital trace concern who constantly clean their browser data are pushed back to algoprithmic feed constantly)

**Comment by codecat** [comment]
In general I think it looks okay, but there's a bit too much stuff I'm really not interested in:

* Threads started in discussions of repositories that I starred (isn't that what "watching" a repository is for?)
* "Recommended for you" is already displayed on the right side too (under "Explore repositories" it shows 3 suggested repositories), so why display it in the feed too?
* Someone that I don't follow forking a repository I didn't star made by a developer that I don't follow (really not sure why this one is even on the feed, to be honest)

Generally, I like suggestions and the feed of starred repositories, but this is just way too much stuff that I don't care about.

**Comment by lukesdm** [comment, marked as answer]
Dear @github-staff, I would like to ask you to also consider the potential impact of those on the other side of this i.e. the users whose actions, without consent, may be either amplified or diminished by an algorithmic feed. For many of us, GitHub is part of our livelihood, and the gamification of it could lead to extra stress we could do without. There are other platforms well-placed for promotion and dissemination of our work - please consider deferring to those, instead of making it a compulsory part of this site.

**Reply by ericoporto** [reply]
I really don't want GSEO or whatever will be the name of SEO for GitHub, life is already stressful as is, I think you touched on the most important topic. Thanks.

**Reply by EduApps-CDG** [reply]
It will probably be like the Facebook feed. Maybe we shouldn't worry about discovery optimizations. For me, it's ok to have something from Facebook, but not everything. I don't want GitHub to become the next TikTok.

**Reply by Stankye** [reply]
I believe this has already happened. Product Hunt and "startup" types have been rallying around star counts for awhile now. Worst offender I've seen as of late was the medusa eCommerce , they also plastered themselves over most of the dev forums.

Idea to counter: Don't use star counts / follow counts but "used by" metrics.

**Reply by adiled** [reply]
Apt way to put it. I like the design improvement but algo magic is a no go. It should be opt-in only if GH team has a desire to experiment with devs' emotions, one can believe however much about their ability to unleash unprecedented ML and tune the UX like never before, but the fact remains, it bastardizes usability and collective productivity in favor of juicing out engagement for a few bucks.

Work on alternate healthier business models!

**Reply by 0956397203** [reply]
👍

**Comment by bearlike** [comment]
Most developers and consumers prefer open-source tools to increase data privacy and transparency. An algorithmic feed might translate to collecting more data, and it doesn’t align with the open-source community. Also from a functionality standpoint, it does not seem different from “`trending repositories`”. Algorithmic feeds could also gamify the platform. This might increase the amount of “low-quality projects” aiming just to be viral.

**Reply by EduApps-CDG** [reply]
But looking for the good part, it is possible to get benefits from the feed such pull requests, stars, followers and donations. This would solve the biggest problem of open source.

**Reply by bearlike** [reply]
That's totally understandable. But can't the same can be achieved without implementing social media like features and minimal data collection?

**Reply by EduApps-CDG** [reply]
There's always a better idea. But I didn't think much about it.

**Reply by Daniel15** [reply]
> Most developers and consumers prefer open-source tools to increase data privacy and transparency.

Github isn't open-source though. Developers that prefer everything to be open-source and avoid proprietary software where possible (often out of principle) tend to use something like Gitlab instead.

**Reply by MichaelWheldon** [reply]
> There's always a better idea. But I didn't think much about it.

Incidentally, you just highlighted why these algorithms are a bad idea with this comment. Well done.

**Comment by xsoheilalizadeh** [comment]
I was expecting to see PRs that are created by my followings!

**Reply by nokiae0106** [reply]
8

**Reply by nokiae0106** [reply]
> 8

**Comment by kwanCCC** [comment]
👍, Me either

**Comment by stefan-it** [comment]
Fully agree! And the old recent activity feature is really missing:

https://github.com/github/feedback/discussions/13150

**Comment by Keiku** [comment]
I'm getting star notifications from users I'm not currently following, but I'm not really interested. I hope that the displayed list will remain the same as before, leaving only batches such as stars.

**Comment by sntwandy** [comment]
I completely disagree with you, I like the new feed, it is more global interaction.

**Comment by mpalaourg** [comment]
Hello, 
I don't want to contribute any more noise, as most of my concerns about the `For You` tab were covered by the previous comments (e.g. ,muting users/events, filter events to be shown).  One more "problem", however, is that for me the `Following` and `For you` tabs are the same at a percentage above 95%. If I didn't click at this discussion I wouldn't understand the purpose of the new tab and I could have dismissed a nice idea of GitHub, to get rid of the extra noise.

tl;dr Somewhat good idea, but need some fixes about the content displayed!

**Comment by madhums** [comment]
Clearly different users are interested in different things, why not give a choice to the user to set what needs to be displayed on the home page or a combination(s) of them?

- Feed (with, without personalisation and chronological)
- Issues
- PR's
- ?

**Reply by TeoTwawki** [reply]
yes, please.. 

In fact it just hit me some other site I happen to use every day for my day job has the ability to add and remove elements and position them where I please to optimize my workflow. Its all proprietary stuff you can' see without logging in and I don't think they'd be ok with my screenshotting it, unfortunately. But I'd love to reclaim the 50% of the screen feed is taking up for something I will actually use.

**Comment by dylmye** [comment]
I don't understand the problem it's trying to solve, if it has one? Also I'm confused why it wasn't rolled out through 'feature preview' to allow users who don't want it to turn it off, it would be good to have some consistency in providing beta features :/

**Comment by shellscape** [comment]
**We don't need a curated feed in Github. This is a technical site, not Instagram. STOP IT.**

**Comment by pbasov** [comment]
What's next, Stories? Github is NOT a social network, it's a code repo and ticket management system. Stop.

**Comment by Olyno** [comment]
I see some people unhappy with this feature, because according to them, Github is not a social network.

Github is by definition a social network:

> group of people linked by common tastes, interests.

Also, adding a news feed allows many things such as the discovery of new projects BY TASTE and not by trend, discovery of new developers that we could sponsor, new features for a project that we have bookmarked, and so on!

So please, stop being petty and judging the new. Instead, try to be constructive and bring ideas for changes rather than putting down a new initiative/project.

**Reply by pbasov** [reply]
You're looking at the feature through rose-tinted glasses. Any algorithmic feed promotes gamification, reverse-engineering and other SEO-type bull**** to promote yourself.

>  Instead, try to be constructive and bring ideas for changes

The idea is simple, remove it before it's too late. We don't need influencers, we don't need hashtags, we don't need stories, we don't need algo-feeds. What we need is to protect the purity of the open-source community, where the best projects end up on the top, because they are well-built and useful, not because a "growth-hacker" spent 40 hours/week to game the algorithm.

**Reply by Olyno** [reply]
For the moment Github has only improved and favored the dev environment, I don't have to wear rose-tinted glasses as you say to see it. If you are not able to see what Github has done in the last months/years, I can't help you.
Maybe you're right, and this is just marketing, but it's still a useful feature that I wouldn't mind using once implemented correctly. It brings many advantages.
If now your only argument is to promote gamification, it becomes very sad to discuss with you.

**Reply by Scoppio** [reply]
Yes, it is a social network, but it is also our livelihoods, and we would very much appreciate that our livelihoods isnt turn into a glorified twitter. Thank you.

**Reply by Olyno** [reply]
Why not an option to enable/disable this feed? That way people who want to get involved in new projects can, and those who don't would not be bothered?

**Reply by AgentOak** [reply]
Somewhat related, if you want to find new projects to get involved in there is the Explore page available which already has interest-based recommendations as well as trends.

**Reply by Olyno** [reply]
I'm not sure trends are interest-based recommendations, but yeah the Explore page should be something to consider in my case, thank you!

**Reply by Siccity** [reply]
If GitHub is a social network then I dread the day they decide to add private group chats, instant-messaging and status messages

**Reply by Olyno** [reply]
Status messages already exist on the profile 😅

**Comment by PranavBhattarai** [comment]
The next thing we know is, M$ will introduce Github ads to make our life more **CONNECTED**. 

Sooner or later. Mark my words.
Amen.

**Comment by was07** [comment]
From my perspective I see GitHub as a productive place, definitely not like a social media. The 'for you' page makes GitHub feel like one. We want only the activity that matters.

Also, is there an option so when I visit github.com it shows me the 'following' tab not the 'for you' tab?

**Comment by jacksonisiah** [comment]
I hate the new trend of "tailored experiences", but if GitHub is going to force a completely useless feed, at least give the option to be able to remove people and repositories that we don't care about. I see cards for things I have not been involved with for a long time and do not care to see it.

**Comment by Siccity** [comment]
Please stop piling junk on this site. You already added enough stuff that nobody needs. It's distracting, time consuming, and worsens the site

**Comment by Mehgugs** [comment]
The new for you feed is less relevant than before. Social media for programmers already exists via subreddits and services like discord, gitter, telegram,... 

I want a feed that shows me activity concretely based on what I'm watching or subscribed to. I do not need extremely popular repositories like sphinx or vscode recommended to me! Stars are a dreadful metric on this website because they say very little about whether a repository is useful to an individual. The more you have the more general your repository's audience is: the less relevance it has to a specific user.

**Comment by madhums** [comment]
I see something new on the dashboard but there isn't an option to remove feed (yet) (And it's a little irrelevant to show "get started on github" to someone who has been using it actively for more than 13 years)

<img width="400" alt="Screenshot 2022-04-24 at 10 22 15" src="https://user-images.githubusercontent.com/123815/164967400-f8a489c4-5267-4d1a-9aa1-c3498b224b2b.png">

**Comment by chrisblatchley** [comment]
hey there just adding my agreement to this thread.

**Comment by Siccity** [comment]
The thing that made GitHub great was that it was concise. Recently a lot of garbage features have been introduced that do nothing but clutter and distract. The feed is only useful for marketing. GitHub should **not** be yet another marketing tool.

**Reply by alanxone** [reply]
I'm afraid they are turning it into this.

Open-source projects in big techs is actually a product strategy, especially M/A/G nowadays are trying to tie the customers within their own ecosystem (yes, you could use some of them elsewhere, but will fall into lots of pains in production eventually.)

Microsoft may expect to make use of the Github to promote its own tech someday, e.g. showing them more often in the Feeds, or with higher ranking.

**Comment by payrim** [comment]
update: they ruined github
the explore and for you are just redundant and this promotion will hit github really bad
i do not like gitlab and some of their practices are not open source for me but looks like the community gotta make a decision if this goes.

**Comment by benjamin-bader** [comment]
Please let me go back to just seeing the events for people and repos I've explicitly followed; this newsfeed is useless to me.  I don't come to Github for social networking or to kill time, I come here to get work done and learn things.  Engagement with a newsfeed is a pretty poor choice of metric to optimize IMO.

**Comment by MichaelWheldon** [comment]
Imagine having all the money in the world and owning an incredible product like Github and one of the first things you do is to go absolute gutter trash levels of "innovation" with a LinkedIn style feed.

Imagine the minds at work and their complete lack of awareness or tact.

**Comment by FrostyX** [comment]
I can't remember how the GitHub dashboard looked before the feed was introduced and how it looked throughout the various feed iterations. I just remember noticing that it is more and more frustrating and useless. At the time of writing this comment, it got to a point where I am interested in the top-left 10% of the screen and my brain shuts out everything else. The useful area is so small that it could fit on a mobile phone screen without scrolling.

When I open GitHub, I almost always want to do one of:

- See my repositories
- See my organizations
- Create a new repository

Why? Because I set out to work on a particular thing so I need to find the project that I want to work on and find an issue or PR in it. I don't want to see all my issues or PRs from all projects because there is a lot of them and they are irrelevant at that time.

Once a year I want to:

- See all my issues or PRs because I forgot in what project is the thing that I care about 
- See all my issues or PRs to ping them if there is no progress 

What I never ever ever ever needed:

- To see the entire feed. But in particular:
    - Who pushed where - We have pull requests, don't we? So we already get this information from notifications
    - Who forked what - Unless it is my project, why should I care? It doesn't mean anything
- The only thing in the feed that at least remotely makes any sense is
    - The slight dopamine hits from seeing that somebody starred my project
    - When somebody that I follow stars a project because there is a minor chance that the project might be interesting to me

What is useless beyond imagination:

- Trending repositories - Even though they seem to be personalized they are still absolutely not interesting. At this moment, I see 3 recommended projects that I don't care about written in 3 programming languages that I don't know and don't care about. The most frustrating thing is that this section has more screen space than the top-left section with my own repositories
- Latest changes - Latest changes of what exactly? Is it a GitHub changelog? Yeah, I don't care. Let alone every time I open GitHub. Just show me one button somewhere
- Explore repositories - I have never used it, and I never will, but it gives me much more interesting results than trending repositories, so ... okay, I don't mind it

So in summary, from the whole 1920x1080 screen, I use

- 7 buttons in the top-left (links to my top repositories)
- Two buttons in the middle of the top panel (Pull requests and Issues)
- The button with my face in the top-right

The UX is just bad, there is no way about it.

**Reply by FrostyX** [reply]
It is worth saying that the idea of a dashboard is a good one and I would like to use it. However, it would have to show information that is useful. The current version of the dashboard is so far removed from that, that I am changing my GitHub bookmark to simply point me to my profile page and ignore the dashboard entirely. And I will probably re-write my profile README to serve as a dashboard. And that's just sad.

**Comment by benjamin-bader** [comment]
Please let me turn off the right-hand sidebar, the one featuring "Trending Repositories".  I actively dislike it, it's distracting, and doesn't assist with the things I actually use Github for (work, OSS maintenance, etc).  If I want to waste time scrolling I'll use Twitter, tyvm.

**Comment by mipmip** [comment]
This new feed is so studid!

**Comment by TheMikeyRoss** [comment]
Why do I care who followed who? why do I need to see that? Why is it taking up so much space in my feed?

**Comment by stefan-it** [comment]
So good news is, that old feed is still available: https://github.com/dashboard-feed

Why is GitHub not listening to their users?
